pkg/kv/repository.go

```diff
@@ -145,7 +145,7 @@ func (cer *Repository) CreateKey(key *model.KeyEntity) (*model.KeyEntity, error)
 
 func (cer *Repository) DeleteKey(key string) error {
 	//bundle DB operations
-	dbOps := func() error {
+	dbOps := func(tx *db.TxConnection) error {
 		//delete all cache entities which were using a value of this key
 		if err := cer.CacheDep.Invalidate().WithKey(key).Exec(false); err != nil {
```

Comment:
this should happen at the end of the DB query execution...

DB.queries have to use `tx` instead of `cer.Conn`